build/android.mk

```diff
@@ -209,29 +216,42 @@ ifeq ($(FAT_BINARY),y)
 ANDROID_LIB_BUILD =
 ANDROID_THIRDPARTY_STAMPS =
 
-# Example: $(eval $(call generate-abi,xcsoar,armeabi-v7a,ANDROID7))
+# Example: $(eval $(call generate-abi,xcsoar,armeabi-v7a,ANDROID7),n) ("n" sets ANDROID_LEGACY)
 define generate-abi
 
 ANDROID_LIB_BUILD += $$(ANDROID_BUILD)/lib/$(2)/lib$(1).so
 
-$$(ANDROID_BUILD)/lib/$(2)/lib$(1).so: $$(OUT)/$(3)/$$(XCSOAR_ABI)/bin/lib$(1).so | $$(ANDROID_BUILD)/lib/$(2)/dirstamp
+OUT_BUILD_DIR := $(3)
+ifeq ($(4),y)
+OUT_BUILD_DIR := $$(OUT_BUILD_DIR)_LEGACY
```

Comment:
This creates a whole new output directory - this is rather surprising, because it can be expected that output is found in `output/$(TARGET)`. Below this, there are directories `dbg` and `opt` for different build options which cause an ABI change. Let's integrate it at this level.

Reply:
Changed to one output directory per architecture.
However it is necessary not only to distinguish the ABI directory but also the third-party directory because the configure runs must run once for API 21, the other time with android-16 platform and API 12.
That does not make the build system easier nor prettier.